tests/test_provision.py

```diff
@@ -28,11 +28,13 @@ def test_dynamic_topology(tmpdir):
 """)
     # prepare config
     app_path = os.path.join(root_dir.strpath, "..")
-    res_args = dict(topology="10_controller,2_compute")
 
+    provision.TopologyArgument.settings_dir = app_path
```

Comment:
Setting the "settings_dir" for the TopologyArgument is required but hidden from user. Can we move settings_dir as an optional argument to the constructor?